js/plugin/RoutingPathQuality.js

```diff
@@ -60,6 +60,159 @@ BR.RoutingPathQuality = L.Control.extend({
+                        // values rendering the special values moot
+                        min: 0,
+                        max: 1,
+                        discreteStrokes: true,
+                    },
+                    valueFunction: (function () {
+                        let cache = [];
+                        return function (latLng) {
+                            var feature = latLng.feature;
+                            if (!feature.wayTags) {
+                                return 1.0;
+                            } else if (cache[feature.wayTags]) {
+                                return cache[feature.wayTags];
+                            }
+                            let data = new URLSearchParams(feature.wayTags.replace(/\s+/g, '&')); // eslint-disable-line compat/compat
+                            let surface = null;
+                            switch (data.get('surface')) {
+                                case 'paved':
+                                    surface = 0.8;
+                                    break;
+                                case 'asphalt':
+                                case 'concrete':
+                                    surface = 1;
+                                    break;
+                                case 'concrete:lanes':
+                                case 'concrete:plates':
+                                    surface = 0.6;
+                                case 'sett':
+                                case 'gravel':
+                                case 'pebblestone':
+                                    surface = 0.5;
+                                    break;
+                                case 'paving_stones':
+                                case 'compacted':
+                                case 'fine_gravel':
+                                    surface = 0.7;
+                                    break;
+                                case 'cobblestone':
+                                case 'dirt':
+                                case 'grass':
+                                    surface = 0.2;
+                                    break;
+                                case 'unhewn_cobblestone':
+                                    surface = 0.01;
+                                    break;
+                                case 'ground':
+                                case 'earth':
+                                    surface = 0.3;
+                                    break;
+                                case 'mud':
+                                case 'sand':
+                                    surface = 0.01;
+                                    break;
+                                case null:
+                                    break;
+                                default:
+                                    console.warn('unhandled surface type', data.get('surface'));
```

Comment:
Please comment out console warning, as with a selected choice of values there will always be warnings without any actionable implication.